vignettes/forcats.Rmd

```diff
@@ -44,17 +43,59 @@ To do so, we can use the function `fct_infreq()`:
 
 ```{r fct-infreq-hair}
 #| fig.alt: >
-#|   The A bar chart of hair color, now ordered so that the least
+#|   The bar chart of hair color, now ordered so that the least
 #|   frequent colours come first and the most frequent colors come last.
 #|   This makes it easy to see that the most common hair color is none 
-#|   (~35), followed by brown (~18), then black (~12).
-ggplot(starwars, aes(x = fct_infreq(hair_color))) + 
-  geom_bar() + 
-  coord_flip()
+#|   (~35), followed by brown (~18), then black (~12). Surprisingly,
+#|   NAs are at the top of the graph, even though there are ~5 NAs and 
+#|   other colors have smaller values.
+ggplot(starwars, aes(y = fct_infreq(hair_color))) + 
+  geom_bar()
 ```
 
 Note that `fct_infreq()` it automatically puts NA at the top, even though that doesn't have the smallest number of entries.
 
+Its's a little surprising that the `NA` bar isn't ordered by frequency.
+To understand why, we need to make a brief digression to discuss `NA`s in values vs `NA`s in levels
+
+## `NAs`s in levels and values
+
+There are two ways to represent a missing value in a factor:
+
+-   You can include it in the values of the factor; it does not appear in the levels and `is.na()` reports it as missing. This is the usual way to encode a missing value in a factor:
+
+```{r}
+f <- factor(c("x", "y", NA))
+levels(f)
+is.na(f)
+```
+
+-   You can include in the levels of the factor and `is.na()` does not report it as missing. This requires a little more work to create:
+
+```{r}
+f <- factor(c("x", "y", NA), exclude = NULL)
```

Comment:
Is there a way to say "don't exclude NAs" without specifying `exclude = NULL`? If not, I think it would be good to add a sentence about why we brought `NULL` into play all of a sudden, like something about "this works because in R `NULL`s are ..."